File: Projekt/Backend/system-kliniki-medycznej/src/main/java/org/back/systemklinikimedycznej/appointment/controllers/dto/FinishAppointmentRequest.java
package org.back.systemklinikimedycznej.appointment.controllers.dto;

import org.back.systemklinikimedycznej.patient.controllers.dto.CreatePatientDiseaseRequest;
import org.back.systemklinikimedycznej.prescription.dto.CreatePrescriptionRequest;

import java.util.List;
import java.util.UUID;

public record FinishAppointmentRequest(
        UUID appointmentId,
        String diagnosis,
        List<CreatePrescriptionRequest> appointmentPrescriptions,
        List<CreatePatientDiseaseRequest> patientDiseases
) {

}

File: Projekt/Frontend/system-kliniki-medycznej/src/app/features/appointment/model/finish-appointment-request.ts
import {CreatePrescriptionRequest} from '../../prescriptions/model/create-prescription-request';
import {CreatePatientDiseaseRequest} from '../../patient-disease/create-patient-disease-request';

export interface FinishAppointmentRequest {
  appointmentId: string;
  diagnosis: string;
  appointmentPrescriptions: CreatePrescriptionRequest[];
  patientDiseases: CreatePatientDiseaseRequest[];
}
